src/mpl/src/gpu/mpl_gpu_ze.c

```diff
@@ -1690,19 +1687,20 @@ static int remove_stale_sender_cache(const void *ptr, uint64_t mem_id, int local
             goto fn_fail;
         }
 
-        HASH_FIND(hh, ipc_cache_tracked[local_dev_id], &mem_id, sizeof(uint64_t), cache_entry);
-        if (cache_entry) {
-            free_ipc_handle_cache(cache_entry);
-            HASH_DELETE(hh, ipc_cache_tracked[local_dev_id], cache_entry);
-            MPL_free(cache_entry);
-        }
-
         HASH_FIND(hh, mem_id_cache, &ptr, sizeof(void *), memid_entry);
         if (memid_entry) {
-            if (memid_entry->mem_id != mem_id || memid_entry->dev_id != local_dev_id) {
-                goto fn_fail;
-            }
             HASH_DELETE(hh, mem_id_cache, memid_entry);
+
+            HASH_FIND(hh, ipc_cache_tracked[memid_entry->dev_id], &mem_id, sizeof(uint64_t),
+                      cache_entry);
+            if (cache_entry) {
+                /* call putIpcHandle only if this ptr is still valid */
+                if (memid_entry->mem_id == mem_id && memid_entry->dev_id == local_dev_id) {
+                    free_ipc_handle_cache(cache_entry);
+                }
```

Comment:
there's nothing we can do about it when the free hooks aren't invoked, but do we know the consequence for not putting the ipc handles when we are done with them?

Reply:
I talked to the level zero team about this, basically If zeMemPutIpcHandle is not called, any actions performed by that call are eventually done by zeMemFree.

Reply:
Thanks, I'm going to merge after a quick build test on Aurora.